js/components/portal-dashboard/multiple-choice-answer.tsx

```diff
@@ -29,24 +37,50 @@ export default class MultipleChoiceAnswer extends PureComponent<IProps> {
   }
 
   renderFullAnswer() {
-    const { question, answer } = this.props;
-    const choices = question.get("choices") || [];
-    const studentChoices = answer.get("selectedChoices");
-    return (
-      <div data-cy="multiple-choice-answers">
-        {
-          choices.size > 0 ? choices.map((choice: any) =>
-            <Choice key={choice.get("id")} choice={choice} correctAnswerDefined={question.get("scored")}
-              selected={studentChoices.some((studentChoice: any) => studentChoice.get("id") === choice.get("id"))}
-            />
-          )  : "Question doesn't have any choices"
-        }
-      </div>
-    );
+    const { question, answer, answers, students, inQuestionDetailsPanel } = this.props;
+    const choices = question.get("choices") || List();
+
+    // Aggregated class responses
+    if (answers && students) {
+      return (
+        <MultipleChoiceChoices
+          answers={answers}
+          choices={choices}
+          inQuestionDetailsPanel={inQuestionDetailsPanel}
+          question={question}
+          showStats={true}
+          students={students}
+        />
+      );
+    }
+
+    // Individual student response
+    if (answer) {
+      return (
+        <MultipleChoiceChoices
+          choices={choices}
+          inQuestionDetailsPanel={inQuestionDetailsPanel}
+          question={question}
+          studentAnswer={answer}
+          showStats={false}
+        />
+      );
+    }
+
+    return null;
   }
 
   render() {
-    const { showFullAnswer } = this.props;
+    const { showFullAnswer = true, answers, students } = this.props;
```

Comment:
Cool.  For some reason I never realized you could do that in TypeScript destructuring with optional variables.